pkg/orchestrator/transformer/job.go

```diff
@@ -16,39 +18,87 @@ func IDGenerator(_ context.Context, job *models.Job) error {
-			if !job.IsLongRunning() {
-				if task.Timeouts.TotalTimeout <= 0 {
-					task.Timeouts.TotalTimeout = int64(defaults.TotalTimeout.Seconds())
-				}
-				if task.Timeouts.ExecutionTimeout <= 0 {
-					task.Timeouts.ExecutionTimeout = int64(defaults.ExecutionTimeout.Seconds())
-				}
-			}
-
-			// only apply default queue timeout to batch and service jobs
-			if job.Type == models.JobTypeBatch || job.Type == models.JobTypeService {
-				if task.Timeouts.GetQueueTimeout() <= 0 {
-					task.Timeouts.QueueTimeout = int64(defaults.QueueTimeout.Seconds())
-				}
-			}
+		switch job.Type {
+		case models.JobTypeBatch:
+			applyBatchJobDefaults(defaults.Batch, job)
+		case models.JobTypeOps:
+			applyBatchJobDefaults(defaults.Ops, job)
+		case models.JobTypeService:
+			applyLongRunningJobDefaults(defaults.Service, job)
+		case models.JobTypeDaemon:
+			applyLongRunningJobDefaults(defaults.Daemon, job)
+		default:
+			return fmt.Errorf("unknown job type: %s", job.Type)
 		}
 
 		return nil
 	}
 	return JobFn(f)
 }
 
+func applyBatchJobDefaults(defaults types.BatchJobDefaultsConfig, job *models.Job) {
+	if job.Priority == 0 {
+		job.Priority = defaults.Priority
+	}
+	for _, task := range job.Tasks {
+		applyBatchTaskDefaults(defaults.Task, task)
+	}
+}
+
+func applyBatchTaskDefaults(defaults types.BatchTaskDefaultConfig, task *models.Task) {
+	if task.ResourcesConfig.CPU == "" {
+		task.ResourcesConfig.CPU = defaults.Resources.CPU
+	}
+	if task.ResourcesConfig.Memory == "" {
+		task.ResourcesConfig.Memory = defaults.Resources.Memory
+	}
+	if task.ResourcesConfig.Disk == "" {
+		task.ResourcesConfig.Disk = defaults.Resources.Disk
+	}
+	if task.ResourcesConfig.GPU == "" {
+		task.ResourcesConfig.GPU = defaults.Resources.GPU
+	}
+	if task.Publisher.IsEmpty() {
+		task.Publisher.Type = defaults.Publisher.Type
+	}
+	if task.Timeouts.ExecutionTimeout <= 0 {
+		task.Timeouts.ExecutionTimeout = int64(time.Duration(defaults.Timeouts.ExecutionTimeout).Seconds())
+	}
+	if task.Timeouts.TotalTimeout <= 0 {
+		task.Timeouts.TotalTimeout = int64(time.Duration(defaults.Timeouts.TotalTimeout).Seconds())
+	}
+}
+
+func applyLongRunningJobDefaults(defaults types.LongRunningJobDefaultsConfig, job *models.Job) {
+	if job.Priority == 0 {
+		job.Priority = defaults.Priority
+	}
+	for _, task := range job.Tasks {
+		applyLongRunningTaskDefaults(defaults.Task, task)
+	}
+}
+
+func applyLongRunningTaskDefaults(defaults types.LongRunningTaskDefaultConfig, task *models.Task) {
+	if task.ResourcesConfig.CPU == "" {
+		task.ResourcesConfig.CPU = defaults.Resources.CPU
+	}
+	if task.ResourcesConfig.Memory == "" {
+		task.ResourcesConfig.Memory = defaults.Resources.Memory
+	}
+	if task.ResourcesConfig.Disk == "" {
+		task.ResourcesConfig.Disk = defaults.Resources.Disk
+	}
+	if task.ResourcesConfig.GPU == "" {
+		task.ResourcesConfig.GPU = defaults.Resources.GPU
+	}
+	if task.Publisher.IsEmpty() {
+		task.Publisher.Type = defaults.Publisher.Type
+	}
+}
```

Comment:
To validate my understanding, long running jobs should have a default publisher, right?

Reply:
Will adjust.